crates/uv-pep508/src/marker/tree.rs

```diff
@@ -695,12 +697,14 @@ impl MarkerTree {
     #[allow(clippy::needless_pass_by_value)]
     pub fn and(&mut self, tree: MarkerTree) {
         self.0 = INTERNER.lock().and(self.0, tree.0);
```

Comment:
Is it better to hold the lock and re-use it in `simplify`, or re-`lock()` as I'm doing here?

Reply:
Probably re-use it, these are pretty much only used by a single thread.